sources/@roots/bud-compiler/src/service.tsx

```diff
@@ -214,32 +214,31 @@
         error: StatsError,
       ): ErrorWithSourceFile | StatsError => {
         let file: SourceFile[`file`] | undefined
+        let module: undefined | Webpack.StatsModule
 
-        const moduleIdent = error.moduleId ?? error.moduleName
+        const ident = error.moduleId ?? error.moduleName
 
         /**
          * In a perfect world webpack plugins would use the
          * `nameForCondition` property to identify the module.
          */
-        let module = this.compilationStats.children
-          .flatMap(child => child?.modules)
-          .find(
-            module =>
-              module?.id === moduleIdent || module?.name === moduleIdent,
-          )
-
-        /**
-         * If the module is not found, we try to parse the error message
-         */
-        if (!moduleIdent) {
-          const stylelintExtracted = error.message.match(
-            /file:\/\/(.*)\x07(.*)\x1B]8;;/,
-          )
-
-          if (stylelintExtracted?.[1]) {
+        if (ident) {
+          module = this.compilationStats.children
+            .flatMap(child => child?.modules)
+            .find(module => [module?.id, module?.name].includes(ident))
+
+          /**
+           * If the module is not found, we try to parse the error message
+           */
+        } else {
+          const styleError = error.message
+            .split(`\n`)?.[1]
+            ?.match(/file:\/\/(.*)\x07(.*)\x1B]8;;/)
```

Comment:
This change specifically addresses the `js/polynomial-redos` problem being highlighted, since it is possible that it could be a performance regression if you had a large number of build errors emitted by stylelint. The fix is to only look at the first line of the message body, since that's where we'll find the file path. Additionally, this code only runs if the error is not shaped normally and is not executed for the majority of errors (in fact it is never executed unless the user has installed @roots/bud-stylelint). There are enough checks in place as is so I'm closing the alert.